app/src/main/java/com/github/swent/echo/data/room/RoomLocalDataSource.kt

```diff
@@ -0,0 +1,104 @@
+package com.github.swent.echo.data.room
+
+import com.github.swent.echo.data.model.Association
+import com.github.swent.echo.data.model.Event
+import com.github.swent.echo.data.model.Tag
+import com.github.swent.echo.data.model.UserProfile
+import com.github.swent.echo.data.repository.datasources.LocalDataSource
+import com.github.swent.echo.data.room.entity.AssociationRoom
+import com.github.swent.echo.data.room.entity.EventRoom
+import com.github.swent.echo.data.room.entity.EventTagCrossRef
+import com.github.swent.echo.data.room.entity.TagRoom
+import com.github.swent.echo.data.room.entity.UserProfileRoom
+import com.github.swent.echo.data.room.entity.UserProfileTagCrossRef
+import com.github.swent.echo.data.room.entity.toTagList
+import com.github.swent.echo.data.room.entity.toTagRoomList
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class RoomLocalDataSource @Inject constructor(db: AppDatabase) : LocalDataSource {
+
+    private val associationDao = db.associationDao()
+    private val eventDao = db.eventDao()
+    private val tagDao = db.tagDao()
+    private val userProfileDao = db.userProfileDao()
+
+    override suspend fun getAssociation(associationId: String): Association? {
+        return associationDao.get(associationId)?.toAssociation()
+    }
+
+    override suspend fun setAssociation(association: Association) {
+        associationDao.insert(AssociationRoom(association))
+    }
+
+    override suspend fun getAllAssociations(): List<Association> {
+        return associationDao.getAll().map { it.toAssociation() }
+    }
+
+    override suspend fun setAssociations(associations: List<Association>) {
+        associationDao.insertAll(associations.map { AssociationRoom(it) })
+    }
+
+    override suspend fun getEvent(eventId: String): Event? {
+        return eventDao.get(eventId)?.toEvent()
+    }
+
+    override suspend fun setEvent(event: Event) {
+        val crossRefs = event.tags.map { EventTagCrossRef(event.eventId, it.tagId) }
+
+        eventDao.insert(EventRoom(event))
+        eventDao.insertEventTagCrossRegs(crossRefs)
+        event.organizer?.let { associationDao.insert(AssociationRoom(it)) }
+        tagDao.insertAll(event.tags.toTagRoomList())
```

Comment:
Good point, I'll change that just in case, but as all tests pass (also the one where I set an event with tags and then retrieve it back), I feel like this isn't really an issue when using room.